sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -21,6 +22,85 @@
     from azure.core.pipeline import PipelineRequest, PipelineResponse
 
 
+class _HttpChallenge(object):
+    """ Represents a parsed HTTP WWW-Authentication Bearer challenge from a server. """
+
+    def __init__(self, request_uri, challenge):
+        self.source_authority = self._get_request_authority(request_uri)
+        self.source_uri = request_uri
+        self._parameters = {}
+
+        # Get the scheme of the challenge and remove it from the challenge string
+        trimmed_challenge = self._get_challenge_parts(challenge)
+        split_challenge = trimmed_challenge.split(" ", 1)
+        self.scheme = split_challenge[0]
+        trimmed_challenge = split_challenge[1]
+
+        # Split trimmed challenge into name=value pairs; these pairs are expected to be split by either commas or spaces
+        # Values may be surrounded by quotes, which are stripped here
+        separator = "," if "," in trimmed_challenge else " "
+        for item in trimmed_challenge.split(separator):
+            # process name=value pairs
+            comps = item.split("=")
+            if len(comps) == 2:
+                key = comps[0].strip(' "')
+                value = comps[1].strip(' "')
+                if key:
+                    self._parameters[key] = value
+
+        if not self._parameters:
+            raise ValueError("Invalid challenge parameters")
+
+        # Challenge must specify authorization or authorization_uri
+        if "authorization" not in self._parameters and "authorization_uri" not in self._parameters:
+            raise ValueError("Invalid challenge parameters")
```

Comment:
I would add a comment specifying why the challenge parameters were invalid (which keys I/we expected)